content/help/ops/security/root-transition/index.md

```diff
@@ -0,0 +1,191 @@
+
+If you are currently using the mutual TLS feature in Istio with self-signed certificates,
+please follow the procedure and check whether you will be affected.
+
+## Root transition procedure
+
+1. Check when the root certificate expires:
+
+Download this [script](TODO) on a machine that has `kubectl` access to the cluster.
+
+{{< text bash>}}
+$ ./root-transition.sh check
+...
+===YOU HAVE 30 DAYS BEFORE THE ROOT CERT EXPIRES!=====
+{{< /text >}}
+
+Execute the remainder of the steps prior to root certificate expiration to avoid system outages.
+
+1. Execute a root certificate transition:
+
+During the transition, the Envoy sidecars may be hot-restarted. This may have some impact on your traffic.
+Please refer to
+[Envoy hot restart](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/hot_restart#arch-overview-hot-restart)
+and read [this](https://blog.envoyproxy.io/envoy-hot-restart-1d16b14555b5)
+blog post for more details.
+
+{{< warning >}}
+If your Pilot does not have a Envoy sidecar, consider installing Envoy sidecar for your Pilot.
+Because the Pilot has issue using the old root certificate to verify the new workload certificates.
+This may cause disconnection between Pilot and Envoy.
+Please see the [here](check-if-pilot-has-an-envoy-sidecar) for how to check.
+The [Istio upgrade guide](/docs/setup/kubernetes/upgrade/steps/)
+by default installs Pilot with Envoy sidecar.
+{{< /warning >}}
+
+{{< text bash>}}
+$ ./root_transition.sh transition
+create new ca cert, with trust domain as  cluster.abc
+Wed Jun  5 19:11:15 PDT 2019 delete old ca secret
+secret "istio-ca-secret" deleted
+Wed Jun  5 19:11:15 PDT 2019 create new ca secret
+secret/istio-ca-secret created
+pod "istio-citadel-8574b88bcd-j7v2d" deleted
+Wed Jun  5 19:11:18 PDT 2019 restarted Citadel, checking status
+NAME                             READY     STATUS    RESTARTS   AGE
+istio-citadel-8574b88bcd-l2g74   1/1       Running   0          3s
+New root certificate:
+Certificate:
+    Data:
+        ...
+        Validity
+            Not Before: Jun  6 03:24:43 2019 GMT
+            Not After : Jun  3 03:24:43 2029 GMT
+        Subject: O = " cluster.abc"
+        ...
+{{< /text >}}
+
+1. Verify the new workload certificates are generated:
+
+{{< text bash>}}
+$ ./root_transition.sh check
+...
+===YOU HAVE 3650 DAYS BEFORE THE ROOT CERT EXPIRES!=====
+Checking the current root CA certificate is propagated to all the Istio-managed workload secrets in the cluster.
+Root cert MD5 is 8fa8229ab89122edba73706e49a55e4c
+Checking namespace: default
+  Secret default.istio.default is updated.
+  Secret default.istio.sleep is updated.
+Checking namespace: istio-system
+  Secret istio-system.istio.default is updated.
+  ...
+------All Istio keys and certificates are updated in secret!
+{{< /text >}}
+
+If this command fails, wait a minute and run the command again.
+It takes time for Citadel to propagate the certificates.
+
+1. Update to Istio 1.1.8:
```

Comment:
I'm wondering should we make this step separate from the main `Root transition procedure`? Put it in a prerequisite section which is required before user start the transition procedure, and make it super clear that user should first verify the upgrade itself works as expected for their cluster and start the transition in a clean state?

The reason is it seems we are coupling two potential high risk task (upgrade istio and root transition) to the same task, user may follow the task and do them together without proper verifying the result of each step, and end up with something goes wrong in their cluster, but not sure if it's caused by upgrading Istio or root cert transition.

Otherwise the worst case is we end up with a lot of customer issues but we don't know if it's caused by upgrading istio or by root cert transition.

@incfly @wenchenglu any thoughts?

Reply:
For many customers, this avoids two hot-restarts of Envoy, so it reduces system transition time. Root transition is low-risk as we tested. Also, previous 1.1 track upgrades are smooth and 1.1.8 upgrade is relatively small, so I don't think it's high risk. I think your thought makes some sense, there may be customers want to do upgrade first. I can put that into a section in troubleshooting.